### DISCUSSION TITLE
Replacing a tag in markdown with iframe embed - issue

### DISCUSSION THREAD

**Discussion by landed1** [remarkjs/remark#1389] [Q&A]
I am using a custom plugin to traverse a markdown file  
`{{<youtube>}}` with `proper youtube iframe` code.

The issue is that an empty set of p tags is created instead of the youtube embed code. If I use `type: "string"` then I can see the string with the correct videoId. So why is using `type: "html"` as shown below causing the deletion. I have checked but this seems to be happening inside the package not something Next or React is doing. 

Thank you.

```import { visit } from "unist-util-visit";

function remarkYouTubeEmbed() {
  return (tree) => {
    visit(tree, "text", (node, index, parent) => {
      const youtubeRegex = /{{< youtube (\S+) >}}/;
      const match = youtubeRegex.exec(node.value);
      if (match) {
        const videoId = match[1];
        parent.children[index] = {
          type: "html",
          value: `<iframe width="560" height="315" src="https://www.youtube.com/embed/${videoId}" ></iframe>`,
        };
        //console.log("youtubeRegex match", videoId);
      }
    });
  };
}

export default remarkYouTubeEmbed;
```

**Comment by ChristianMurphy** [comment]
Welcome @landed1! 👋 
Could you put your code example into a code block so it is readable?
````
```js
{{code goes here}}
```
````

In addition consider sharing a runnable example so we can see what you see.
Using either:
* https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/remark-with-vite
* https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/remark-rehype-with-vite
* or https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-vite
as a starting point
